base-image/build.sh

```diff
@@ -8,6 +8,6 @@ mkdir -p $BUILD_DIR
 cp -r $SOURCE_DIR/* $BUILD_DIR/
 cd $BUILD_DIR
 source poky/oe-init-build-env build
+sed -e s~#{MACHINE_NAME}~"$TARGET_MACHINE"~g conf/local.conf.tpl > conf/local.conf
```

Comment:
No need to do this. you can invoke bitbake like so: `MACHINE=generic-armv6 bitbake core-image-minimal`